src/client/Client.cc

```diff
@@ -7362,6 +7372,8 @@ int Client::_readdir_get_frag(dir_result_t *dirp)
   req->head.args.readdir.flags = CEPH_READDIR_REPLY_BITFLAGS;
   if (dirp->last_name.length()) {
     req->path2.set_path(dirp->last_name.c_str());
+  } else if (dirp->hash_order()) {
+    req->head.args.readdir.offset_hash = dirp->offset_high();
```

Comment:
Is offset_hash getting converted to network byte order somewhere?  I guess it should be the same mechanism as the rest of the request header, but I can't see where it's heappning